src/test/getarg_tests.cpp

```diff
@@ -200,7 +200,7 @@ BOOST_AUTO_TEST_CASE(logargs)
     const auto okaylog_negbool = std::make_pair("-okaylog-negbool", ArgsManager::ALLOW_BOOL);
     const auto okaylog = std::make_pair("-okaylog", ArgsManager::ALLOW_ANY);
     const auto dontlog = std::make_pair("-dontlog", ArgsManager::ALLOW_ANY | ArgsManager::SENSITIVE);
```

Comment:
> test/getarg_tests.cpp:218: error: in "getarg_tests/logargs": check str.find("Command-line arg: okaylog-bool=\"\"") != std::string::npos has failed
> test/getarg_tests.cpp:219: error: in "getarg_tests/logargs": check str.find("Command-line arg: okaylog-negbool=false") != std::string::npos has failed
> test/getarg_tests.cpp:220: error: in "getarg_tests/logargs": check str.find("Command-line arg: okaylog=\"public\"") != std::string::npos has failed
> test/getarg_tests.cpp:221: error: in "getarg_tests/logargs": check str.find("dontlog=****") != std::string::npos has failed

My tests passed in the first 2 commits, but this commit gives me these errors locally. It looks like the bug comes from my `debug.log` having something that it should not. My struggle is I can't really find my debug log...I checked out some resources (Andrew Chow's/bitcoin wiki), but none seems to mention where to locate this file. 

https://gist.github.com/fjahr/2cd23ad743a2ddfd4eed957274beca0f does include an example command `$ cat ~/Library/Application\ Support/Bitcoin/regtest/debug.log` but this doesn't work for me locally. Any tips? @amitiuttarwar   @fjahr  🙏

Reply:
re: https://github.com/bitcoin/bitcoin/pull/18926#discussion_r427038185

> My tests passed in the first 2 commits, but this commit gives me these errors locally. It looks like the bug comes from my `debug.log` having something that it should not.

That's right, the problem is happening because the arguments are being set in the `argman` instance not `gArgs` but line 214 still says `gArgs.LogArgs()` not `argsman.LogArgs()`.

> My struggle is I can't really find my debug log...I checked out some resources (Andrew Chow's/bitcoin wiki), but none seems to mention where to locate this file.

I believe it's written to a temporary directory so you could check TMPDIR, but most convenient way to see logs from tests is:

```
src/test/test_bitcoin -l test_suite -t getarg_tests -- DEBUG_LOG_OUT
```

Also, in my previous comment https://github.com/bitcoin/bitcoin/pull/18926#discussion_r423913184, I did link to a working updated copy of the test. If you want to apply and test those changes locally you can do that with:

```c++
git fetch https://github.com/ryanofsky/bitcoin review.18926.1-edit
git checkout FETCH_HEAD src/test/getarg_tests.cpp
make -C src test/test_bitcoin
src/test/test_bitcoin -l test_suite -t getarg_tests -- DEBUG_LOG_OUT
```

Reply:
ahhhh thank you! I avoided line 214 in my sed command, because it would modify `LogArgs()` to `Loargsman()`, but I overlooked the use of the global manager in there. I will look what sed command I can use to avoid changing only a specific location of line.

I feel like if I commited your solution, my scripted-diff commit on its own is still failing the test...this makes my commits depend on each other, instead of working modularly.

Reply:
You can use 

```
$ echo 'gArgs.LogArgs()' | sed -e 's/\<gArgs\>/____/g'
____.LogArgs()

Reply:
> I feel like if I commited your solution

To be sure, suggestion was just to compare against for debugging, not to commit

Reply:
> You can use
> 
> ```
> $ echo 'gArgs.LogArgs()' | sed -e 's/\<gArgs\>/____/g'
> ____.LogArgs()
> ```

Thanks Marco, I tried running this but it is not changing anything from my end. Is there any setup I needed to do? Are you using gnu-sed? This is what I tried.
`sed -i -e 's/\<gArgs\>/argsman/g' src/test/getarg_tests.cpp` 

I did receive a similar suggestion from my [post](https://stackoverflow.com/questions/61957896/sed-global-substitution-how-to-ignore-a-particular-range-of-positions-in-one?stw=2) on S/O, but somehow mine is not working 😂

Reply:
> > I feel like if I commited your solution
> 
> To be sure, suggestion was just to compare against for debugging, not to commit

Totally. Thanks again for your suggestion :)

Reply:
> > You can use
> > ```
> > $ echo 'gArgs.LogArgs()' | sed -e 's/\<gArgs\>/____/g'
> > ____.LogArgs()
> > ```
> 
> Thanks Marco, I tried running this but it is not changing anything from my end. Is there any setup I needed to do? Are you using gnu-sed? This is what I tried.
> `sed -i -e 's/\<gArgs\>/argsman/g' src/test/getarg_tests.cpp`
> 
> I did receive a similar suggestion from my [post](https://stackoverflow.com/questions/61957896/sed-global-substitution-how-to-ignore-a-particular-range-of-positions-in-one?stw=2) on S/O, but somehow mine is not working 😂

Maybe I should look more into the regex itself actually

Reply:
Yes, you will need to use gsed. Our scripted diffs don't support macos sed, only Ubuntu sed.